plugins/Toolbox/src/Toolbox.py

```diff
@@ -0,0 +1,609 @@
+            self.setIsDownloading(False)
+            if self._download_plugin_reply:
+                self._download_plugin_reply.downloadProgress.disconnect(self._onDownloadProgress)
+                self._download_plugin_reply.abort()
+                self._download_plugin_reply = None
+            return
+
+        if reply.error() == QNetworkReply.HostNotFoundError:
+            Logger.log("w", "Unable to reach server.")
+            return
+
+        if reply.operation() == QNetworkAccessManager.GetOperation:
+            # TODO: In the future use the following to build any model from any
+            # request. Right now this doesn't work because the packages request
+            # is also responsible for populating other models.
+            # for type, url in self._request_urls.items():
+            #     if reply.url() == url:
+            #         try:
+            #             json_data = json.loads(bytes(reply.readAll()).decode("utf-8"))
+            #
+            #             # Check for errors:
+            #             if "errors" in json_data:
+            #                 for error in json_data["errors"]:
+            #                     Logger.log("e", "%s", error["title"])
+            #                 return
+            #
+            #             # Create model and apply metadata:
+            #             if not self._models[type]:
+            #                 Logger.log("e", "Could not find the %s model.", type)
+            #                 break
+            #             self._metadata[type] = json_data["data"]
+            #             self._models[type].setMetadata(self._metadata[type])
+            #             self.metadataChanged.emit()
+            #             self.setViewPage("overview")
+            #             return
+            #         except json.decoder.JSONDecodeError:
+            #             Logger.log("w", "Toolbox: Received invalid JSON for %s.", type)
+            #             break
+
+            if reply.url() == self._request_urls["packages"]:
+                try:
+                    json_data = json.loads(bytes(reply.readAll()).decode("utf-8"))
+
+                    # Check for errors:
+                    if "errors" in json_data:
+                        for error in json_data["errors"]:
+                            Logger.log("e", "%s", error["title"])
+                        return
+
+                    # Create packages model with all packages:
+                    if not self._models["packages"]:
+                        self._models["packages"] = PackagesModel(self)
+                    self._metadata["packages"] = json_data["data"]
+                    self._models["packages"].setMetadata(self._metadata["packages"])
+                    self.metadataChanged.emit()
+
+                    # Create authors model with all authors:
+                    if not self._models["authors"]:
+                        self._models["authors"] = AuthorsModel()
```

Comment:
This is always constructed in the `__init__` function, so it will never hit this `if` statement.

Reply:
Yup. Artifact from before I decided "why not just create it on init?"